gsoc-2024-project-list.md

```diff
@@ -0,0 +1,218 @@
+* [Eliminate Maven Plugin Usage](#project-name-eliminate-maven-plugin-usage)
+
+***
+
+### Project Name: Java 21 Language Feature Support
+
+**Skills required**: Java, basic understanding of testing principles, basic understanding of static analysis
+
+**Project goal**: Support static analysis of Java 21 language features
+
+**Project size**: large
+
+**Mentors**: Nick Mancuso, Roman Ivanov
+
+**Description**:
+
+Developers are enthusiastic about leveraging Java's latest language features, which offer more powerful, declarative,
+and expressive code; these features
+include [unnamed variables](https://openjdk.org/jeps/443), [record patterns](https://openjdk.org/jeps/440),
+and [string templates](https://openjdk.org/jeps/430). However, Checkstyle currently lacks robust support for these
+features. This project aims to bridge this gap by updating existing checks and potentially introducing new ones to
+ensure thorough coverage of Java 21 syntax and conventions proposed by the JEPs associated with these language
+advancements. The objective is to deliver comprehensive support for the new language features through revising check
+modules, exhaustive testing, and detailed documentation updates. This effort not only aligns Checkstyle with
+cutting-edge best practices in the Java community but also contributes to the project's ongoing evolution.
+
+**Deliverables**:
+
+- Analysis of new language features and possible static analysis coverage
+- Check module updates
+- Documentation improvements
+- Establish best practices for new language features
+
+***
+
+### Project Name: Automated Website Generation
+
+**Skills required**: Java, basic understanding of testing principles, technical writing, continuous integration
+
+**Project goal**: organize documentation and automate its maintenance
+
+**Project size**: medium
+
+**Mentors**: Roman Ivanov, Nick Mancuso, Vyom Yadav
+
+**Description**:
+
+This project is designed to tackle the persistent challenge of
+maintaining accurate and current documentation in our dynamic development environment. Acknowledging the limitations of
+manual documentation processes, this initiative introduces automation to streamline content creation, with a focus on
+ensuring consistent formats and robust verification checks. The project's goal is to provide users with reliable,
+standardized, and regularly updated information while equipping contributors with templates and automated tools to
+simplify the incorporation of details for new modules. By elevating documentation practices, this project aligns with
+industry best practices, fostering clarity for both users and contributors within the Checkstyle project.
+
+**Deliverables**:
+
+- Resolution of edge cases in documentation generation
+- Extend and make consistent all check usage examples
+- Reduce/eliminate manual documentation updates for examples
+- Introduce checks to ensure that all configuration options are covered in examples
```

Comment:
moved to https://github.com/checkstyle/checkstyle/wiki/Checkstyle-GSoC-2024-Project-Ideas#project-name-automated-website-generation